datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -726,6 +747,8 @@ impl ExecutionPlan for HashJoinExec {
                     context.clone(),
                     join_metrics.clone(),
                     reservation,
+                    need_produce_result_in_final(self.join_type),
+                    1,
```

Comment:
was confuse about  why pass 1 to `right_input_partitions`, after it pass to `running_threads_counter` maybe change var name from `right_input_partitions` to `probe_threads_number` something is better 🤔

Reply:
Sounds reasonable -- I've renamed both arguments (in collect function and JoinLeftData constructor) to `probe_thread_count` and `probe_thread_counter` respectively